src/components/TokenCard/TokenCard.tsx

```diff
@@ -58,6 +58,57 @@ type Props = {
   showRedirectModal?: (to: string) => void;
 };
 
+const BuyLink = ({
+  chainId,
+  active,
+  className,
+  to,
+  children,
+  network,
+  showRedirectModal,
+}: {
+  chainId: number;
+  active: boolean;
+  className: string;
+  to: string;
+  children: React.ReactNode;
+  network: number;
+  showRedirectModal?: (to: string) => void;
+}) => {
+  const changeNetwork = useCallback(
+    (network) => {
+      console.log("network", network, chainId);
+
+      if (network === chainId) {
+        return;
+      }
+      if (!active) {
```

Comment:
Maybe some comment here to explain why it needs setTimeout if `!active`

Reply:
addded